javascript/lib/lib.js

```diff
@@ -0,0 +1,103 @@
+const jws = require('jws');
+const { Base64 } = require('js-base64');
+const SignatureError = require("./error.js");
+const Headers = require("./headers.js");
+
+let requireArg = (arg, name) => {
+  if (!arg) throw new Error(`missing argument ${name}`);
+  return arg;
+};
+
+let buildV2SigningPayload = ({ method, path, headers, body }) => {
```

Comment:
Is there a `buildV1SigningPayload`? Consider using namespaces/modules for version specific nomenclature.